### ISSUE TITLE
Terraform Scripts for forming PrivateLink with dedicated cluster.

### ISSUE THREAD

**Issue by VipulZopSmart** [confluentinc/terraform-provider-confluent#81]
Hi @linouk23 , hope you're doing well.

I've created terraform scripts for provisioning of privateLink setup with dedicated cluster.
https://github.com/confluentinc/terraform-provider-confluent/blob/master/examples/configurations/dedicated-privatelink-aws-kafka-acls/main.tf

There's no issue while creating PrivateLink and dedicated cluster but the issue is every time it creates privateLink with different zones - `us2-az2`,`us2-az3`, `us2-az4` or `us2-az1`,`us2-az3`, `us2-az4`. but as per my zone IDs it should be `us2-az1`,`us2-az2`, `us2-az3`. After this when i try to destroy this cluster and then provision it again then it creates with right Zones. 

I'm facing this issue whenever i run this scripts.

**Comment by VipulZopSmart** [comment]
The Zones you can see in this screenshot are different from the inputs that i've given:
```
subnets_to_privatelink = {
   "usw2-az1" = "subnet-xyz",
   "usw2-az2" = "subnet-abc",
   "usw2-az3" = "subnet-def",
 }
```
 
<img width="716" alt="Screenshot 2022-08-08 at 7 06 32 PM" src="https://user-images.githubusercontent.com/77278140/183444680-98fff78e-38ef-48ed-abf8-58b7ec7d4027.png">


```
It's failing this with error.
│ Error: creating EC2 VPC Endpoint (com.amazonaws.vpce.us-west-2.vpce-svc-0a28dd62aa81eaaf5): InvalidParameter: The VPC endpoint service com.amazonaws.vpce.us-west-2.vpce-svc-0a28dd62aa81eaaf5 does not support the availability zone of the subnet: subnet-abc.
│       status code: 400, request id: 1f92ff66-17fe-4919-a7d8-2be5852a7cb1
│
│   with aws_vpc_endpoint.privatelink,
│   on main.tf line 141, in resource "aws_vpc_endpoint" "privatelink":
│  141: resource "aws_vpc_endpoint" "privatelink" {
```

**Comment by linouk23** [comment]
Thanks for opening an issue @VipulZopSmart!

This issue looks the same as #80 (see https://github.com/confluentinc/terraform-provider-confluent/issues/80#issuecomment-1204842851).

TLDR you might want to add `zones = keys(var.subnets_to_privatelink)` to fix it.

**Comment by VipulZopSmart** [comment]
Thanks a lot, will verify and update here. I took help from the example and i think it was mentioned in that example.

**Comment by linouk23** [comment]
Update: we updated an example to include `zones            = keys(var.subnets_to_privatelink)` in our latest 1.1.0 [release](https://github.com/confluentinc/terraform-provider-confluent/blob/master/CHANGELOG.md#110-august-9-2022).

**Comment by VipulZopSmart** [comment]
I've provided zones like above, now it's taking so much time to create privateLink with the right zones. @linouk23

**Comment by VipulZopSmart** [comment]
@linouk23 It used to create privateLink in around 5-7 mins before, but now it's taking around 50+ mins, and still creating privateLink. How much time it takes generally, also i've checked the plan everything looks fine to me.

**Comment by linouk23** [comment]
@VipulZopSmart it might be an issue on our backend, could you email your networkID and orgID to `cflt-tf-access@confluent.io` so we could take a look at it?

**Comment by VipulZopSmart** [comment]
Hi @linouk23, even after you email i'm getting into the same issue again. Could you please debugging the issue?
